### PR TITLE
refactor: remove legacy 'usd'-related ledger transaction properties

### PR DESCRIPTION
**Pull request by vindard**
## Description

**_Note: only merge after #2184_** 

This PR removes the `usd`, `feeUsd` and `fee` properties from ledger transactions. The information from these is now encoded in the `satsAmount`-related properties on transactions.

### CODE REVIEW

File: test/unit/domain/wallets/tx-history.spec.ts [hunk 1]
```diff
@@ -223,13 +236,15 @@ describe("translates ledger txs to wallet txs", () => {
...
       const settlementAmount = toSats(100000)
-      const usd = 20
+      const centsAmount = toCents(2000)
 
       const txnsArgs = {
         walletId: crypto.randomUUID() as WalletId,
         settlementAmount,
-        usd,
-        feeUsd,
+        centsAmount,
```
Comment (dolcalmi) [inline test/unit/domain/wallets/tx-history.spec.ts:244]: shouldn't this add sats amount and fee?

File: test/unit/domain/wallets/tx-history.spec.ts [hunk 2]
```diff
@@ -181,13 +190,15 @@ describe("translates ledger txs to wallet txs", () => {
...
```
Comment (dolcalmi) [inline test/unit/domain/wallets/tx-history.spec.ts:198]: same here

File: test/unit/domain/wallets/tx-history.spec.ts [hunk 3]
```diff
@@ -33,15 +35,21 @@ describe("translates ledger txs to wallet txs", () => {
   const ledgerTxnsInputs = ({
     walletId,
     settlementAmount,
-    usd,
+    centsAmount,
```
Comment (dolcalmi) [inline test/unit/domain/wallets/tx-history.spec.ts:38]: shouldn't this add satsAmount?

File: src/services/ledger/index.types.d.ts [hunk 4]
```diff
@@ -76,20 +70,26 @@ type SendAmountsMetadata = {
...
     hash: OnChainTxHash
     payee_addresses: OnChainAddress[]
     sendAll: boolean
   }
 
+type NonIntraledgerLedgerMetadata = LedgerMetadata & {
+  usd: number // to be removed when "centAmount" takes over
+  fee: Satoshis
+  feeUsd: number // to be removed when "centFee" takes over
+}
```
Comment (dolcalmi) [inline src/services/ledger/index.types.d.ts:91]: not sure to understand this. aren't we removing this fields?

File: src/domain/wallets/tx-history.ts [hunk 5]
```diff
@@ -317,7 +337,9 @@ const displayCurrencyPerBaseUnitFromAmounts = ({
 }: {
   displayAmountAsNumber: number
   settlementAmountInBaseAsNumber: number
-}): number =>
-  settlementAmountInBaseAsNumber === 0
+}): number => {
+  const usdDisplayAmountAsNumber = Number((displayAmountAsNumber / 100).toFixed(2))
```
Comment (dolcalmi) [inline src/domain/wallets/tx-history.ts:341]: why to rename it to usd?

**Review by dolcalmi — CHANGES_REQUESTED** [review]
looks good, just minor changes/comments

File: src/domain/wallets/tx-history.ts (hunk 5 above)
Comment (vindard) [reply src/domain/wallets/tx-history.ts]: Fixed here, I was trying to capture that it wasn't "cents" unit https://github.com/GaloyMoney/galoy/pull/2191/commits/44b0e9871afdd7cdabf79be770117e1af72d6118

File: test/unit/domain/wallets/tx-history.spec.ts (hunk 3 above)
Comment (vindard) [reply test/unit/domain/wallets/tx-history.spec.ts:38]: https://github.com/GaloyMoney/galoy/pull/2191#discussion_r1084196607

File: src/services/ledger/index.types.d.ts (hunk 4 above)
Comment (vindard) [reply src/services/ledger/index.types.d.ts:92]: I haven't finished removing them from our Admin transaction types as yet. I actually had to add back some additional (temporary) provisions for this after this comment here https://github.com/GaloyMoney/galoy/pull/2191/commits/7d9ad41413e417dda5a4f2096bcc558dc34873d6

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**